packages/markdown/remark/src/remark-content-rel-image-error.ts

```diff
@@ -0,0 +1,47 @@
+import type { Image } from 'mdast';
+import { visit } from 'unist-util-visit';
+import type { VFile } from 'vfile';
+
+/**
+ * `src/content/` does not support relative image paths.
+ * This plugin throws an error if any are found
+ */
+export default function remarkContentRelImageError() {
+	return (tree: any, vfile: VFile) => {
+		if (!vfile.path.includes('content/')) return;
+
+		const relImagePaths = new Set<string>();
+		visit(tree, 'image', function raiseError(node: Image) {
+			if (isRelativePath(node.url)) {
+				relImagePaths.add(node.url);
+			}
+		});
+		if (relImagePaths.size === 0) return;
+
+		const errorMessage =
+			`Relative image paths are not support in the content/ directory. Please update to absolute paths:\n` +
+			[...relImagePaths].map((path) => JSON.stringify(path)).join(',\n');
+
+		// Throw raw string to use `astro:markdown` default formatting
+		throw errorMessage;
+	};
+}
+
+// Following utils taken from `packages/astro/src/core/path.ts`:
+
+function isRelativePath(path: string) {
+	return startsWithDotDotSlash(path) || startsWithDotSlash(path);
+}
+
+function startsWithDotDotSlash(path: string) {
+	const c1 = path[0];
+	const c2 = path[1];
+	const c3 = path[2];
+	return c1 === '.' && c2 === '.' && c3 === '/';
```

Comment:
FYI: `path.startsWith('../')` (ES6, Node.js ≥ 4).

Reply:
This is pulled directly from our core utility, so don't want to modify!

Reply:
The reason we do it this way is that it's a very tiny perf optimization!